### PR TITLE
Add View API to enable the stencil buffer

### PR DESCRIPTION
**Pull request by bejado**
Will update RELEASE_NOTES when the Material API is added, otherwise at the moment there's no way to use the stencil buffer.

### CODE REVIEW

File: filament/include/filament/View.h [hunk 1]
```diff
@@ -637,6 +637,33 @@ class UTILS_PUBLIC View : public FilamentAPI {
...
+     * The stencil buffer is an 8-bit, per-fragment unsigned integer stored alongside the depth
+     * buffer. The stencil buffer is cleared at the beginning of a frame and discarded after the
+     * color pass.
+     *
+     * Each fragment's stencil value is set during rasterization by specifying stencil operations on
+     * a Material. The stencil buffer can be used as a mask for later rendering by setting a
+     * Material's stencil comparison function and reference value. Fragments that don't pass the
+     * stencil test are then discarded.
+     *
+     * Post-processing must be enabled in order to use the stencil buffer.
```
Comment (pixelflinger) [inline filament/include/filament/View.h:652]: why is that a requirement?

File: filament/src/RendererUtils.cpp [hunk 2]
```diff
@@ -92,9 +94,15 @@ FrameGraphId<FrameGraphTexture> RendererUtils::colorPass(
...
+                    // clear newly allocated depth/stencil buffers, regardless of given clear flags
                     clearDepthFlags = TargetBufferFlags::DEPTH;
-                    data.depth = builder.createTexture("Depth Buffer", {
+                    clearStencilFlags = config.enabledStencilBuffer ?
+                            TargetBufferFlags::STENCIL : TargetBufferFlags::NONE;
+                    const char* const name = config.enabledStencilBuffer ?
+                             "Depth/Stencil Buffer" : "Depth Buffer";
+                    TextureFormat format = config.enabledStencilBuffer ?
+                            TextureFormat::DEPTH32F_STENCIL8 : TextureFormat::DEPTH32F;
+                    data.depth = builder.createTexture(name, {
```
Comment (pixelflinger) [inline filament/src/RendererUtils.cpp:105]: make sure this works with transmission, which reuses the depth buffer (also transmission + MSAA)

File: filament/src/View.cpp [hunk 3]
```diff
@@ -275,6 +275,14 @@ bool View::isScreenSpaceRefractionEnabled() const noexcept {
     return upcast(this)->isScreenSpaceRefractionEnabled();
 }
 
+void View::setStencilBufferEnabled(bool enabled) noexcept {
+    upcast(this)->setStencilBufferEnabled(enabled);
+}
+
+bool View::getStencilBufferEnabled() const noexcept {
+    return upcast(this)->getStencilBufferEnabled();
+}
```
Comment (pixelflinger) [inline filament/src/View.cpp:284]: don't forget the java and js bindings

**Review by pixelflinger — APPROVED** [review]
looks pretty good, just had a couple questions

File: filament/include/filament/View.h (hunk 1 above)
Comment (bejado) [reply filament/include/filament/View.h:652]: At the moment, the default render target does not allocate a Stencil buffer. Maybe we should.
Technically the requirement is either post-processing OR MSAA must be enabled.

File: filament/src/RendererUtils.cpp (hunk 2 above)
Comment (bejado) [reply filament/src/RendererUtils.cpp:105]: Good idea, I'll check

File: filament/include/filament/View.h (hunk 1 above)
Comment (pixelflinger) [reply filament/include/filament/View.h:652]: Ah right. Maybe we should add a flag to the swapchain creation to specify it needs to support stencil? And say that it's a requirement if post-process is not used?

File: filament/src/View.cpp (hunk 3 above)
Comment (bejado) [reply filament/src/View.cpp:284]: Yes :) I'll add them in a separate PR once we have Material APIs to actually do something interesting with the stencil buffer.

File: filament/include/filament/View.h (hunk 1 above)
Comment (bejado) [reply filament/include/filament/View.h:652]: Yeah that's a good idea

**Pull request merged**